doc/design/request_response_communication.md

```diff
@@ -0,0 +1,102 @@
+# Request/Response Communication
+
+## Summary
+
+Beside the publish-subscribe messaging pattern initially supported by iceoryx,
+the request-response pattern is also widely used in the field where iceoryx is headed.
+Since we aim for a good integration with ROS 2, this is also necessary to support its
+`Service` communication method which is often used to control robots.
+
+While publish-subscribe is a data distribution pattern which continuously delivers
+updates on its data, request-response is a remote procedure call and task distribution pattern
+which delivers data only on demand. This means there is one or several clients
+requesting a server to perform a task.
+
+In iceoryx, the client and server will be built on top of the same building blocks
+we already use for publish-subscribe, like the `ChunkSender` and `ChunkReceiver`.
+
+## Terminology
+
+| Name              | Description                                              |
+| :---------------- | :------------------------------------------------------- |
+| Request-Response  | communication pattern which delivers data on demand      |
+| Publish-Subscribe | communication pattern which continuously delivers updates on its data |
+| RPC               | a remote procedure call to offload task to a server      |
+| Client            | an entity invoking RPCs                                  |
+| Server            | an entity processing RPCs                                |
+
+## Design
+
+### Considerations
+
+Client and server shall be able to be used in combination with the `WaitSet` and `Listener`
+to provide a means of event based communication instead of polling.
+
+Like with publish-subscribe, the `ServiceDescription` will be used to connect clients with a corresponding server.
+
+In order to support asynchronous requests, a sequence ID should be part of each request which will be copied to the corresponding response.
+
+### Solution
+
+This is an overview of the untyped `Client` and `Server` classes.
+
+![simple class diagram](diagrams/request_response_overview_class.svg)
+
+The `Client` and `Server` are reusing the `ChunkSender` and `ChunkReceiver` building blocks. The `Client` uses a `ChunkSender` to send requests and a `ChunkReceiver` to get the responses while the `Server` uses a `ChunkReceiver` to get the requests and a `ChunkReceiver` to send the responses.
+
+#### Typed API
+
+![typed API](diagrams/request_response_typed_api.svg)
+
+#### Untyped API
+
+![untyped API](diagrams/request_response_untyped_api.svg)
+
+#### Client Port
+
+![client port](diagrams/request_response_client_port.svg)
```

Comment:
Could you move the `puml` and `svg` files into a folder called `request_response`? I did the same in #784 Otherwise this folder will become quite large soon ;-)